src/Entity/OgMembership.php

```diff
@@ -295,13 +295,22 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
   /**
    * {@inheritdoc}
    */
-  public function PreSave(EntityStorageInterface $storage) {
+  public function preSave(EntityStorageInterface $storage) {
 
     if (!$this->getFieldName()) {
       $this->setFieldName(OgGroupAudienceHelper::DEFAULT_FIELD);
     }
 
-    parent::PreSave($storage);
+    // Check the value directly rather than using the entity, if there is one.
+    // This will watch actual empty values and '0'.
+    if (!$this->get('uid')->target_id) {
+      // Throw a generic logic exception as this will likely get caught in
+      // \Drupal\Core\Entity\Sql\SqlContentEntityStorage::save and turned in an
+      // EntityStorageException anyway.
+      throw new \LogicException('OG membership can not be created for an empty or anonymous user.');
```

Comment:
Yeah :) Also made a note above about why it would throw an `EntityStorageException` 👍